osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/networking/DlmsPushNotificationDecoder.java

```diff
@@ -130,28 +315,80 @@ private DlmsPushNotification buildPushNotification() {
+                this.builder.withTriggerType(PUSH_ALARM_TRIGGER);
+            } else {
+                LOGGER.warn("Unknown Push notification not supported. Unable to decode");
+                this.builder.withTriggerType("");
+            }
+        } catch (ProtocolAdapterException e) {
+            throw new UnrecognizedMessageDataException("Error decoding logical name", e);
         }
 
         this.builder.withAlarms(null);
         this.builder.appendBytes(logicalNameBytes);
     }
 
+    private boolean isLogicalNameSmsTrigger(byte[] logicalNameBytes, Protocol protocol) throws ProtocolAdapterException {
+        if (protocol == Protocol.DSMR_4_2_2) {
+            // DSMR4 has specific objects for the different external trigger types for SMS and CSD
+            return Arrays.equals(dlmsObjectConfigDsmr422.getObisForObject(EXTERNAL_TRIGGER_SMS).bytes(),
+                    logicalNameBytes);
+        } else {
+            // SMR5 has one general object for the SMS and CSD triggers
+            return Arrays.equals(dlmsObjectConfigSmr50.getObisForObject(EXTERNAL_TRIGGER).bytes(),
+                    logicalNameBytes);
+        }
+    }
+
+    private boolean isLogicalNameCsdTrigger(byte[] logicalNameBytes, Protocol protocol) throws ProtocolAdapterException {
+        if (protocol == Protocol.DSMR_4_2_2) {
+            // DSMR4 has specific objects for the different external trigger types for SMS and CSD
+            return Arrays.equals(dlmsObjectConfigDsmr422.getObisForObject(EXTERNAL_TRIGGER_CSD).bytes(),
+                    logicalNameBytes);
+        } else {
+            // SMR5 has one general object for the SMS and CSD triggers
+            return Arrays.equals(dlmsObjectConfigSmr50.getObisForObject(EXTERNAL_TRIGGER).bytes(),
+                    logicalNameBytes);
+        }
+    }
+
+    private boolean isLogicalNameSchedulerTrigger(byte[] logicalNameBytes, Protocol protocol) throws ProtocolAdapterException {
+        DlmsObjectConfig dlmsObjectConfig = getDlmsObjectConfigForProtocol(protocol);
+
+        return Arrays.equals(dlmsObjectConfig.getObisForObject(PUSH_SCHEDULER).bytes(), logicalNameBytes) ||
+                Arrays.equals(dlmsObjectConfig.getObisForObject(PUSH_SETUP_SCHEDULER).bytes(), logicalNameBytes);
+    }
+
+    private boolean isLogicalNameAlarmTrigger(byte[] logicalNameBytes, Protocol protocol) throws ProtocolAdapterException {
+        DlmsObjectConfig dlmsObjectConfig = getDlmsObjectConfigForProtocol(protocol);
+
+        return Arrays.equals(dlmsObjectConfig.getObisForObject(INTERNAL_TRIGGER_ALARM).bytes(), logicalNameBytes) ||
+                Arrays.equals(dlmsObjectConfig.getObisForObject(PUSH_SETUP_ALARM).bytes(), logicalNameBytes);
+    }
+
+    private DlmsObjectConfig getDlmsObjectConfigForProtocol(Protocol protocol) {
+        if (protocol == Protocol.DSMR_4_2_2) {
+            return dlmsObjectConfigDsmr422;
+        } else {
+            return dlmsObjectConfigSmr50;
+        }
+    }
```

Comment:
Multiple methods with the same condition in them... Supports my suggestion to refactor into separate classes for SMR5 and DSMR4

Reply:
Yes it is refactored